src/lib/auth/src/performAuth.js

```diff
@@ -26,22 +23,22 @@ export const createAuthURL = async (authorizationEndpoint, scope = "openid email
 
     return `${authorizationEndpoint}?` + buildUrlEncodedData({
         response_type: "code",
-        client_id: CLIENT_ID,
+        client_id: clientId,
         state,
         scope,
-        redirect_uri: AUTH_CALLBACK_URL,
+        redirect_uri: authCallbackURL,
         code_challenge: await pkceChallengeFromVerifier(verifier),
         code_challenge_method: "S256",
     }).toString();
 };
 
 const DEFAULT_REDIRECT = "/overview";
 
-export const performAuth = async (authorizationEndpoint, scope = "openid email") => {
-    window.location = await createAuthURL(authorizationEndpoint, scope);
+export const performAuth = async (authorizationEndpoint, clientId, authCallbackURL,  scope = "openid email") => {
```

Comment:
im wondering if this function could be wrapped inside a hook, so that it could have access to the state / config provider

```js
export const usePerformAuth = () => {
  // get authorizationEndpoint, clientId, authCallbackURL from config - maybe scope too!
  return (scope = "openid email") => {
    window.location = await createAuthURL(authorizationEndpoint, clientId, authCallbackURL, scope);
  }
}

const performAuth  = usePerformAuth();  // as a hook, this gets access to the config
// now performAuth has the signature (scope) => void;  - it still needs to be called to actually do anything
```

Reply:
I looked it up, it cannot

Reply:
why can it not?

Reply:
it has to be within a node to access it